From 3d3ede000e02ef337ac64467bd4aea561942ccd4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 29 Jan 2021 12:31:13 +0100
Subject: [PATCH 1/1] lsefisystab: short text for EFI_RT_PROPERTIES_TABLE_GUID

UEFI specification 2.8 errata B introduced the EFI_RT_PROPERTIES_TABLE
describing the services available at runtime.

The lsefisystab command is used to display installed EFI configuration
tables. Currently it only shows the GUID but not a short text for the new
table.

Provide a short text for the  EFI_RT_PROPERTIES_TABLE_GUID.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 grub-core/commands/efi/lsefisystab.c | 1 +
 include/grub/efi/api.h               | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/grub-core/commands/efi/lsefisystab.c b/grub-core/commands/efi/lsefisystab.c
index d29188efa..456198e5f 100644
--- a/grub-core/commands/efi/lsefisystab.c
+++ b/grub-core/commands/efi/lsefisystab.c
@@ -47,6 +47,7 @@ static const struct guid_mapping guid_mappings[] =
     { GRUB_EFI_LZMA_CUSTOM_DECOMPRESS_GUID, "LZMA CUSTOM DECOMPRESS"},
     { GRUB_EFI_MEMORY_TYPE_INFORMATION_GUID, "MEMORY TYPE INFO"},
     { GRUB_EFI_MPS_TABLE_GUID, "MPS"},
+    { GRUB_EFI_RT_PROPERTIES_TABLE_GUID, "RT PROPERTIES"},
     { GRUB_EFI_SAL_TABLE_GUID, "SAL"},
     { GRUB_EFI_SMBIOS_TABLE_GUID, "SMBIOS"},
     { GRUB_EFI_SMBIOS3_TABLE_GUID, "SMBIOS3"},
diff --git a/include/grub/efi/api.h b/include/grub/efi/api.h
index 39733585b..34109861a 100644
--- a/include/grub/efi/api.h
+++ b/include/grub/efi/api.h
@@ -344,6 +344,11 @@
     { 0xab, 0xb6, 0x3d, 0xd8, 0x10, 0xdd, 0x8b, 0x23 } \
   }
 
+#define GRUB_EFI_RT_PROPERTIES_TABLE_GUID \
+  { 0xeb66918a, 0x7eef, 0x402a, \
+    { 0x84, 0x2e, 0x93, 0x1d, 0x21, 0xc3, 0x8a, 0xe9 } \
+  }
+
 struct grub_efi_sal_system_table
 {
   grub_uint32_t signature;
-- 
2.30.0

